File: mobile-ionic/src/app/entities/consultoria.ts
import { Endereco } from './endereco';

export class Consultoria{
    id:string = '';
    cnpj:string = '';
    nomeFantasia:string = '';
    razaoSocial:string = '';
    email:string = '';
    nomeContato:string = '';
    telefone:string = '';
    endereco:Endereco;
    areaFoco:string = '';

    /**transforma o json em consultoria */
    static fromJSON(json: ConsultoriaJSON|string): Consultoria{
        //se 
        if(typeof json === 'string'){
            return JSON.parse(json, Consultoria.reviver);
        }else{
            //cria uma instancia de consultoria
            let consultoria = Object.create(Consultoria.prototype);
            
            consultoria.id = json.id;
            consultoria.cnpj = json.cnpj;
            consultoria.nomeFantasia = json.nomeFantasia;
            consultoria.razaoSocial = json.razaoSocial;
            consultoria.email = json.email;
            consultoria.nomeContato = json.nomeContato;
            consultoria.telefone = json.telefone;
            consultoria.areaFoco = json.areaFoco;
            // consultoria.endereco = json.endereco;
            return consultoria;
        }
    }
    static reviver(key: string, value: any):any {
        return key === '' ? Consultoria.fromJSON(value) : value;
    }

}

interface ConsultoriaJSON {
    id:string;
    cnpj:string;
    nomeFantasia:string;
    razaoSocial:string;
    email:string;
    nomeContato:string;
    telefone:string;
    endereco:Endereco;
    areaFoco:string;
}

File: aps04/fontes/src/facul/domain/Consultoria.java
package facul.domain;

import java.io.Serializable;

import javax.xml.bind.annotation.XmlRootElement;

@XmlRootElement
public class Consultoria implements Serializable{

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;
	
	private int id;
	private String cnpj;
	private String nomeFantasia;
	private String razaoSocial;
	private String email;
	private String nomeContato;
	private String telefone;
	private Endereco endereco;
	private String areaFoco;
	
	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public String getCnpj() {
		return cnpj;
	}
	public void setCnpj(String cnpj) {
		this.cnpj = cnpj;
	}
	public String getNomeFantasia() {
		return nomeFantasia;
	}
	public void setNomeFantasia(String nomeFantasia) {
		this.nomeFantasia = nomeFantasia;
	}
	public String getRazaoSocial() {
		return razaoSocial;
	}
	public void setRazaoSocial(String razaoSocial) {
		this.razaoSocial = razaoSocial;
	}
	public String getEmail() {
		return email;
	}
	public void setEmail(String email) {
		this.email = email;
	}
	public String getNomeContato() {
		return nomeContato;
	}
	public void setNomeContato(String nomeContato) {
		this.nomeContato = nomeContato;
	}
	public String getTelefone() {
		return telefone;
	}
	public void setTelefone(String telefone) {
		this.telefone = telefone;
	}
	public Endereco getEndereco() {
		return endereco;
	}
	public void setEndereco(Endereco endereco) {
		this.endereco = endereco;
	}
	public String getAreaFoco() {
		return areaFoco;
	}
	public void setAreaFoco(String areaFoco) {
		this.areaFoco = areaFoco;
	}
	
	public void adicionaConsultoria(Consultoria consultoria) {
		this.id = consultoria.getId();
		this.cnpj = consultoria.getCnpj();
		this.nomeFantasia = consultoria.getNomeFantasia();
		this.razaoSocial = consultoria.getRazaoSocial();
		this.email = consultoria.getEmail();
		this.nomeContato = consultoria.getNomeContato();
		this.telefone = consultoria.getTelefone();
		this.endereco = consultoria.getEndereco();
		this.areaFoco = consultoria.getAreaFoco();
	}
	
	
	
}
